core/src/main/java/io/kestra/core/test/flow/Assertion.java

```diff
@@ -0,0 +1,35 @@
+package io.kestra.core.test.flow;
+
+import io.kestra.core.models.property.Property;
+import jakarta.validation.constraints.NotNull;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.List;
+
+@Getter
+@Builder
+public class Assertion {
+    @NotNull
+    private Property<Object> value;
+
+    private String taskId;
+
+    private Property<String> errorMessage;
+
+    private Property<String> successMessage;
```

Comment:
I'm not sure a successMessage will have any use

Reply:
me neither, but we could have a description instead WDYT ?

Reply:
KISS, an optional error message is enought, if not set we will create one by ourself.
I even wonder if it should be dynamic or not, I'm not sure at assertion time we will have something to make it dynamic or not.